tests/src/core/testqgsvectortileconnection.cpp

```diff
@@ -0,0 +1,77 @@
+/***************************************************************************
+  testqgsvectortileconnection.cpp
+  --------------------------------------
+  Date                 : January 2022
+  Copyright            : (C) 2022 by Alexander Bruy
+  Email                : alexander dot bruy at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgstest.h"
+#include <QObject>
+#include <QString>
+
+//qgis includes...
+#include "qgsapplication.h"
+#include "qgsvectortileconnection.h"
+
+/**
+ * \ingroup UnitTests
+ * This is a unit test for the vector tile provider connection class
+ */
+class TestQgsVectorTileConnection : public QObject
+{
+    Q_OBJECT
+
+  public:
+    TestQgsVectorTileConnection() = default;
+
+  private:
+
+  private slots:
+    void initTestCase();// will be called before the first testfunction is executed.
+    void cleanupTestCase();// will be called after the last testfunction was executed.
+    void init() {} // will be called before each testfunction is executed.
+    void cleanup() {} // will be called after every testfunction.
+
+    void test_encodedUri();
+};
+
+
+void TestQgsVectorTileConnection::initTestCase()
+{
+  // init QGIS's paths - true means that all path will be inited from prefix
+  QgsApplication::init();
+  QgsApplication::initQgis();
+}
+
+void TestQgsVectorTileConnection::cleanupTestCase()
+{
+  QgsApplication::exitQgis();
+}
+
+void TestQgsVectorTileConnection::test_encodedUri()
+{
+  QgsVectorTileProviderConnection::Data conn;
```

Comment:
@alexbruy 

There's already test_qgsvectortile.py:testEncodeDecodeUri which could be extended here instead of adding another cpp test file

Reply:
Ergh wait, I see -- that's testing a different method